tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,884 @@
+                    EndTag etag;
+                    // track the node level nesting of nodes handled in this function only
+                    int level = 1;
+
+                    do { type = xpp->next(); }
+                    while (type != XmlPullParser::END_DOCUMENT && type != XmlPullParser::START_TAG);
+
+                    if (XmlPullParser::END_DOCUMENT == type)
+                        throw makeStringException(ESDL_MANIFEST_ERR, "Manifest file missing root <Manifest> tag");
+
+                    // Eat the start/end tags for <em:Manifest>
+                    lastType = type;
+                    xpp->readStartTag(stag);
+
+                    const char* localName = stag.getLocalName();
+                    const char* uri = stag.getUri();
+                    if (isEmptyString(localName) || !streq(localName, MANIFEST_TAG_MANIFEST))
+                        throw makeStringException(ESDL_MANIFEST_ERR, "Manifest file root tag must be <Manifest>");
+
+                    if (isEmptyString(uri))
+                        throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Manifest file not using expected namespace '%s'", MANIFEST_URI);
+
+                    if (!streq(uri, MANIFEST_URI))
+                        throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Manifest file using incorrect namespace URI '%s', must be '%s'", uri, MANIFEST_URI);
+
+                    // Command line --output-format overrides manifest attribute
+                    if (!isEmptyString(optOutputFormat.get()) && streq(optOutputFormat.get(), "binding"))
+                        isBundle = false;
+                    else
+                    {
+                        const char* bundleAttr = stag.getValue("outputFormat");
+                        if (bundleAttr)
+                        {
+                            if (streq(bundleAttr, "binding"))
+                                isBundle = false;
+                            else if (!streq(bundleAttr, "bundle"))
+                                throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Unknown value '%s' provided for attribute Manifest[@outputFormat]", bundleAttr);
+                        }
+                    }
+                    if (isBundle)
+                        appendStartTag(MANIFEST_TAG_BUNDLE);
+
+                    do
+                    {
+                        type=xpp->next();
+                        switch (type)
+                        {
+                            case XmlPullParser::START_TAG:
+                            {
+                                xpp->readStartTag(stag);
+                                if (isManifestNodeNamed(stag, MANIFEST_TAG_SVC_BINDING))
+                                    type = processServiceBinding(xpp.get(), stag);
+                                else if (isManifestNodeNamed(stag, MANIFEST_TAG_SVC_DEFINITION))
+                                    type = processServiceDefinition(xpp.get(), stag);
+                                else
```

Comment:
I'm inclined to say that its treated like any other unrecognized markup and copied through to the output. This afternoon I'll see if Tony has any thoughts if that would be detrimental to future backward compatibility. I'm also not checking in every loop for elements in the manifest namespace in places where they're unsupported or nonsensical.

Reply:
I would not expect to see any manifest markup in output that is not meant to be a manifest. Manifest markup is not expected anywhere the output will be used. Between omission of markup or failing, I would probably choose failure because we don't know what the expected output should be.